aten/src/ATen/native/cuda/jit_utils.cpp

```diff
@@ -1500,7 +1500,11 @@ NvrtcFunction jit_pwise_function(
     std::stringstream ss;
     ss << *cache_dir << "/";
     ss << kernel_name;
+#ifdef USE_ROCM
+    ss << "_arch" << prop->gcnArchName;
```

Comment:
It shouldn't ever be empty.  I can't think of a suitable fallback if it were empty.

Reply:
potentially it could be empty if the hip implementation is not provided by ROCM, HIP-CPU for instance dosent seam to provide it https://github.com/ROCm/HIP-CPU/blob/3fef321669ac3409e4a58b3c20b0d3a60a50c925/include/hip/hip_types.h#L63 i am also not sure what the behavior is if the hip backend is cuda

However i dont think this is an issue as the hip usage is rocm-hip specific all over the place, the define is called USE_ROCM not USE_HIP. As mentioned i cant think of a suitable fallback either, for the HIP-CPU case it would just not compile and for other cases i could offer an assert if desired.

Reply:
Don't worry about HIP-CPU, that is not a supported pytorch build target.